src/filefuncs.c

```diff
@@ -75,11 +94,27 @@ static char **getfiles(const char *fspec, int *nfound, LOG_INFO* LogInfo) {
 			(*nfound)++;
 			if (alloc) {
 				flist = (char **) Mem_ReAlloc(flist, sizeof(char *) * (*nfound), LogInfo);
+                flist[(*nfound) - 1] = NULL;

```

Comment:
Re commit cafc45b5156470637ee15e654482d4bd10d6c4fe, in `getfiles()`:

`flist[(*nfound) - 1] = NULL;` doesn't work if the precedent call to `flist = Mem_ReAlloc(...)` or `flist = Mem_Malloc(...)` fails. If it fails, then `flist` is NULL and cannot be subset (undefined behavior) and `freeGetfilesEarly()` does not free memory that was previously allocated to `flist` before it was overwritten to NULL.